src/python/pants/python/python_setup.py

```diff
@@ -181,8 +179,23 @@ def resolve_all_constraints_was_set_explicitly(self) -> bool:
     def interpreter_search_paths(self):
         return self.expand_interpreter_search_paths(self.options.interpreter_search_paths)
 
-    @property
+    # Note: only memoized to avoid the deprecation happening multiple times. Change back once the default
+    # changes
+    @memoized_property
     def resolver_version(self) -> ResolverVersion:
+        if self.options.is_default("resolver_version"):
+            warn_or_error(
+                removal_version="2.4.0.dev0",
```

Comment:
Is this intentional? The removal is 2.5.0.dev0 per the language below. 2.4.0.dev0 is only the default flip.

Reply:
Yep, this deprecation is solely for the default. Then in 2.4, we will remove this deprecation and flip the default, and deprecate the option entirely.

Reply:
Hrm - OK. A bit unfortunate assuming the word removal shows up in help output - since that is not true. But thats the infra we have. SG.